docs/controls/src/imageviewer-all/ImageViewerWindow.qml

```diff
@@ -0,0 +1,59 @@
+import QtQuick 2.0
+import QtQuick.Controls 2.4
+import QtQuick.Dialogs 1.2
+
+ApplicationWindow {
+
+    // ...
+
+    function openFileDialog() { fileOpenDialog.open(); }
+    function openAboutDialog() { aboutDialog.open(); }
+
+    visible: true
+    title: qsTr("Image Viewer")
+
+    background: Rectangle {
+        color: "darkGray"
+    }
+
+    Image {
+        id: image
+        anchors.fill: parent
+        fillMode: Image.PreserveAspectFit
+        smooth: true
+        asynchronous: true
+    }
+
+    FileDialog {
+        id: fileOpenDialog
+
+        // ...
+
+        title: "Select an image file"
+        folder: shortcuts.documents
+        nameFilters: [
+            "Image files (*.png *.jpeg *.jpg)",
+        ]
+        onAccepted: {
+            image.source = fileOpenDialog.fileUrl
+        }
```

Comment:
If you want to cut down on the length of the example, you could make both of these expressions one-liners. I'm not really bothered though.

Reply:
done